### DISCUSSION TITLE
Franka Emika support

### DISCUSSION THREAD

**Discussion by fz0707** [visose/Robots#234] [Q&A] [answered]
Hi Vice,

It's Feng, long time no talk!

We have a project at the University of Nottingham focusing on the interaction between humans and Franka Emika. Currently, we are discussing the best platform for our application. I'm curious about the progress of the plugin in terms of supporting real-time interaction with Franka. We have also explored ROS and Moveit as alternatives, but personally, I have a strong preference for Robots since I've been using it for many years.

Best regards,
Feng

**Comment by visose** [comment, marked as answer]
Hi Feng, hope you are doing well. The plugin supports Franka Emika robots, but the built-in functionality is for offline programming, not for real-time interaction. 
For real-time interaction or anything that deviates from the existing functionality, you can use the custom command component.
To create a simulation you will have to use conventional targets though. For example, if you use a custom command that moves the TCP following a vector until it finds some resistance, you will have to simulate this by using a conventional linear motion and stopping at a known position. Then replace the linear motion with the custom command before uploading to the controller.
